app/controllers/comments_controller.rb

```diff
@@ -5,27 +5,34 @@ class CommentsController < ApplicationController
 
   # GET /comments or /comments.json
   def index
+    redirect_back fallback_location: root_url, alert: "You're not authorized for that."
     @comments = Comment.all
   end
 
   # GET /comments/1 or /comments/1.json
-  def show; end
+  def show
+    redirect_back fallback_location: root_url, alert: "You're not authorized for that."
+  end
 
   # GET /comments/new
   def new
     @comment = Comment.new
   end
 
   # GET /comments/1/edit
-  def edit; end
+  def edit
+    redirect_back fallback_location: root_url, alert: "You're not authorized for that."
+  end
 
   # POST /comments or /comments.json
   def create
     @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.conversation = Conversation.find_by(id: params[:comment][:conversation_id])
```

Comment:
It looks like this should already be in `comment_params` after you added `:converstation_id`, which would make this line unnecessary.